lib/Models/StatisticsClicks.php

```diff
@@ -51,4 +53,31 @@ static function getAllForSubscriber(Subscriber $subscriber) {
       ->where('clicks.subscriber_id', $subscriber->id())
       ->orderByAsc('url');
   }
+
+  static function findLatestPerNewsletterBySubscriber(Subscriber $subscriber, DateTime $before, $since_days_ago) {
```

Comment:
Why did you use `DateTime` here? Wouldn't `DateTimeInterface` be better? We wouldn't be limited and could call this with `DateTime` or `DateTimeImmutable`.